### ISSUE TITLE
Change Turbo to an OptIn for third party modules

### ISSUE THREAD

**Issue by bennyborn** [contao/contao#7454]
### Affected version(s)

5.4

### Description

While I understand the idea behind the usage of Turbo I would strongly suggest that this is done on a OptIn basis for third party modules. 

For example in our [Contao Marketing Suite](https://github.com/numero2/contao-marketing-suite) we're doing some inline JavaScript for generating Charts or doing a Health Check. The later does some AJAX requests and is build on the intention that the script execution is stopped as soon as the user navigates somewhere else in the backend. 

Because Turbo is not doing any real navigation, rather than just replacing the content of the current document, the script execution still continues in the background and can lead to some unforseen problems.

No worries regarding our own extensions, we can surely handle that but I'm not sure if there aren't other extensions that might struggle with that concept.

So I would suggest that, besides the Core modules, Turbo should only enabled if the extension explicitly supports it.

**Comment by fritzmg** [comment]
I agree that we should implement a way to handle this better for extension developers. We could make Turbo Opt-In - and then likewise Opt-In all the core back end modules. This way any back end modules (or custom back end controllers) from extensions or the App won't have Turbo enabled by default and thus nothing can break when updating from <5.4 to >=5.4 .

**Comment by fritzmg** [comment]
One initial idea:

* Have a `_enable_turbo` request attribute for back end controllers.
* Have a `'turbo' => true` config option for `BE_MOD` back end module entries.

If both are missing or false, we set `data-turbo="false"` on `<html>` in `be_main`. And we'll set it to `true` for all core back end modules accordingly.

**Comment by bennyborn** [comment]
**Please note:** This will not help if a "non Turbo module" is doing anything asynchronous because as soon as the user switches to a Turbo enabled module the script execution in the background will still be going. So while this won't fix all problems I guess it'd be a first step in the right direction.

**Comment by fritzmg** [comment]
@bennyborn what do you mean? If you are on a page where Turbo is disabled, everything will be executed as a regular request, without Turbo.Drive.

**Comment by bennyborn** [comment]
Yes but if this page does something asynchronous (like an AJAX request) and the user changes to another page where Turbo is not disabled, the script execution in the background will still be going event though you've just "navigated" to another page which can lead to problems if your tring to do e.g. some DOM manipulation after the AJAX stuff.

**Comment by fritzmg** [comment]
I still do not understand. If you are on a page, where Turbo is disabled and you dispatch an AJAX request, that AJAX request will be disregarded when you navigate away.

**Comment by bennyborn** [comment]
Not if you "navigate" to a Turbo enabled page 🤷‍♂️

**Comment by fritzmg** [comment]
I still do not understand what you mean. Can you give an example?

**Comment by bennyborn** [comment]
I'll try to break it down:

1. Lets assume we're in the backend on a menu item that has `data-turbo="false"` and on this page there is an inline JS that performs any task or AJAX request that takes a long time
2. We click on any other menu item that is Turbo enabled and does not have `data-turbo="false"` so the DOM is "replaced" instantly.
3. While the DOM is already replaced, our task from the first step is finished and the script tries to do some DOM manipulation. That will end in an error because the DOM already changed since we "navigated" (not a real navigation since it's only JS) away

**Comment by fritzmg** [comment]
> 1. Lets assume we're in the backend on a menu item that has `data-turbo="false"`

What do you mean by that? `data-turbo="false"` needs to be on the `<html>` attribute. `data-turbo="false"` on the _menu item_ will only disable Turbo Drive for the request of that link. But Turbo will still be enabled on the page itself once loaded of course.

**Comment by bennyborn** [comment]
OK got it. So having a way to set `data-turbo="false"` for the whole page should prevent that problem from occuring in the first place 👍

**Comment by fritzmg** [comment]
Correct, if the response of your back end module contains `<html data-turbo="false">`, then Turbo will be disabled for that whole page. Currently there is no convenient way to achieve this - only via the `outputBackendTemplate` hook or `kernel.response` event where you would have to edit the buffer, which is not ideal.

**Comment by fritzmg** [comment]
@m-vo wdyt about https://github.com/contao/contao/issues/7454#issuecomment-2293417235 ?

**Comment by m-vo** [comment]
> @m-vo wdyt about https://github.com/contao/contao/issues/7454#issuecomment-2293417235 ?

This could be an option, yes. 👍 Although I think I still prefer it to be opt-out just for the change to be noticed. But we'll need to discuss this, I guess.

**Comment by leofeyer** [comment]
As discussed in the call, we want to add a way to opt-out of using Turbo.

To allow the admin to do this, we want to use `attr()` in the `be_main` template, so they can extend it and add `data-turbo="false"` to the `<html>` element.

To allow third-party extension developers to do this, we want to use the response context in the back end and enhance it as outlined in #7469.
